WordPress/Classes/ViewRelated/Site Creation/Site Name/SiteNameViewController.swift

```diff
@@ -29,10 +29,28 @@ class SiteNameViewController: UIViewController {
         setTitleForTraitCollection()
     }
 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        SiteCreationAnalyticsHelper.trackSiteNameViewed()
+    }
+
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         view.becomeFirstResponder()
     }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        if isMovingFromParent {
```

Comment:
I wonder if we should have a cancelled event here at all? This event will be fired when users go back to the "Topic" selection, which is not a cancel action. The design selection screen, as of today, does not have an event that tracks back navigation.

Reply:
The Design selection screen doesn't, but the Site Intent does - and there it's used to actually "Cancel" the view. I agree the name is subpar, and we debated it a little yesterday but landed on keeping the event consistent between Site Intent and Site Name. It really means "The user didn't progress, they went in the opposite direction of progressing."